src/toil/mpcopy.py

```diff
@@ -0,0 +1,62 @@
+from urlparse import urlparse
+import boto
+import math
+import itertools
+
+
+class S3File:
+    def __init__(self, url):
+        self.s3 = boto.connect_s3()
+        self.url = urlparse(url)
+        assert self.url.scheme == 's3'
+        self.key = self.url.path[1:]
+        bucketName = self.url.netloc
+        self.bucket = self.s3.get_bucket(bucketName)
+
+    def copyTo(self, dest, totalSize, part_size=None, headers={}):
+        partSize = part_size or 2**20 * 50  # 50MiB
+
+        # initiate copy
+        upload = dest.bucket.initiate_multipart_upload(dest.key, headers=headers)
+        try:
+            start = 0
+            partIndex = itertools.count()
+            while start < totalSize:
+                end = min(start + partSize-1, totalSize-1)  # bytes are indexed form zero
```

Comment:
Why is partSize decremented by 1? Doesn't this mean we will always be uploading (partSize-1) bytes at a time instead of the specified partSize? Also I think it should be "indexed from zero" vs. "indexed form zero"